### ISSUE TITLE
Breadcrumb for categories is not multistore compatible

### ISSUE THREAD

**Issue by Prestaworks** [PrestaShop/PrestaShop#29631]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add screenshots

The Breadcrumb for categories is not multistore compatible and always get the name from the first store.

The issue is in Category class where id shop is not used.
```
public function getAllParents($idLang = null)
    {
        if (null === $idLang) {
            $idLang = Context::getContext()->language->id;
        }

        $categories = new PrestaShopCollection('Category', $idLang);
        $categories->where('nleft', '<', $this->nleft);
        $categories->where('nright', '>', $this->nright);
        $categories->orderBy('nleft');

        return $categories;
    }
```

### Expected behavior

The expected behavior is that the category name in the breadcrumb is fetched using the correct multistore context.

### Steps to reproduce

1. Create a new multistore, duplicating the default store from fresh install.
2. go to category Clothes, select the new mutlistore and rename category for this multistore.
3. Check front end in both stores and go to sub category Men or Women.
4. The breadcrumb in both stores is still Clothes.

### PrestaShop version(s) where the bug happened

8.0.0

### PHP version(s) where the bug happened

7.3.21

**Comment by hibatallahAouadni** [comment]
Hello @Prestaworks 

Did you encounter this issue with the older version like 1.7.8.6 or 1.7.7.8?
Waiting for your feedback.

Thanks!

**Comment by Prestaworks** [comment]
> Hello @Prestaworks
> 
> Did you encounter this issue with the older version like 1.7.8.6 or 1.7.7.8? Waiting for your feedback.
> 
> Thanks!

Yes, it was present in 1.7.7.5 also (maybe earlier also)

**Comment by AureRita** [comment]
Hi @Prestaworks ,

I reproduce the issue with PrestaShop version 1.7.6.5, I'll add this to the backlog so it can be fixed.

you can see the issue here : 

https://user-images.githubusercontent.com/100699445/190417780-60fa9fce-b4b4-41bd-8aab-53e4e7dcb1f6.mp4

Please be aware that some issues might take a very long time to be resolved. 

Thanks for all the details !

**Comment by eduvega16** [comment]
Hello,

Anyone resolve this problem? 
![image](https://github.com/PrestaShop/PrestaShop/assets/48015700/f9f3aef9-7372-4998-a961-075ca752c76a)

this is the url: https://www.tramasmas.com/fr/pyjamas-courts-homme/1000049374-47774-pijama-hombre-manga-corta-diego-burdeos-8445168173609.html#/31-talla-m

**Comment by Prestaworks** [comment]
This seems to be fixed on PS8

**Comment by hibatallahAouadni** [comment]
@paulnoelcholot could you please try to reproduce it and check if it's fixed on 8.1.x 🙏

**Comment by florine2623** [comment]
It is fixed on branch `8.0.x.`

<img width="1440" alt="Screenshot 2023-10-12 at 09 34 51" src="https://github.com/PrestaShop/PrestaShop/assets/16019289/06a2fbb7-3dec-4306-ab02-8e971b780cf7">

<img width="1227" alt="Screenshot 2023-10-12 at 09 35 12" src="https://github.com/PrestaShop/PrestaShop/assets/16019289/ee9c5eab-dfab-4c3f-8710-b3d5e88dfc5d">
<img width="1437" alt="Screenshot 2023-10-12 at 09 36 50" src="https://github.com/PrestaShop/PrestaShop/assets/16019289/5d7ec1ea-3fd7-4867-9bae-9cba31b30d3a">
<img width="1195" alt="Screenshot 2023-10-12 at 09 36 40" src="https://github.com/PrestaShop/PrestaShop/assets/16019289/846415ab-29c6-41ac-b995-cafc326b3559">
